### DISCUSSION TITLE
can not add record of TXT for same RRset

### DISCUSSION THREAD

**Discussion by zhaohong-ponyai** [PowerDNS/pdns#13962] [Q&A] [answered]
### Short description
i have a zone like '`example.rpz`'
and there is a type of A record as
storage.example.rpz IN A 192.168.0.0.1

and then i want to add a type of TXT record like:
storage.example.rpz IN TXT "region=cn"

and the api response get an error as:
RRset storage.example.rpz. IN TXT: Conflicts with pre-existing RRset

### Environment
powerdns 4.4.1

### Expected behaviour
different type of record can add success 

### Actual behaviour
can not add record of TXT for same rr

is there any limit here?

**Comment by omoerbeek** [comment]
You are using an unsupported version. Please upgrade. See https://repo.powerdns.com

There are very few restrictions, but an important one is that you cannot have a CNAME record together with another type of record for a specific name.

Please show the exact requests you are doing and the full replies, plus all existing records before yo are trying to add a record.

**Reply by zhaohong-ponyai** [reply]
That is, there can't be rr of cname and rr of txt type the same?
or can i fix this problem by upgrade the pdns version ?

**Reply by mnordhoff** [reply, marked as answer]
> That is, there can't be rr of cname and rr of txt type the same?

The DNS protocol doesn't allow it. Upgrading PowerDNS won't change that.

But if you're having a **different** issue, upgrading PowerDNS might fix it or make it easier to debug.

**Comment by mnordhoff** [comment]
`192.168.0.0.1` is an invalid IP address, for what it's worth. It's too long.

**Comment by zhaohong-ponyai** [comment]
> `192.168.0.0.1` is an invalid IP address, for what it's worth. It's too long.
Sorry, I wrote that IP wrong. Actually, it has nothing to do with this IP. Type A records have been added successfully, such as:
storage.example.rpz IN A 192.168.0.1

I am now, add the above records, and then add the following TXT records, it is wrong
storage.example.rpz IN TXT "region=cn"

so my demand that i can also add this TXT for same rr record

**Reply by omoerbeek** [reply]
The records in an rrset set **must** all be of the same type.

It is possible to have multiple rrsets for a name.

We have asked before: please show the *exact* API queries you are doing, plus the contents of the DB before doing the queries. Otherwise we cannot help you.

**Reply by zhaohong-ponyai** [reply]
API is this:
![image](https://github.com/PowerDNS/pdns/assets/20126149/d28af439-49fd-46dc-9944-74acb19de487)
before doing queries
![image](https://github.com/PowerDNS/pdns/assets/20126149/64459730-c686-43f0-a9a1-2f64eb0c2598)

and i want to add a record as 
storage.example.rpz IN TXT "region=cn"

and got the error:
RRset storage.example.rpz. IN TXT: Conflicts with pre-existing RRset

I can 't find anyother solutions, So I had to come to you and see if there was anything I could do. Sorry

**Reply by Habbie** [reply]
> We have asked before: please show the _exact_ API queries you are doing, plus the contents of the DB before doing the queries. Otherwise we cannot help you.

Please do this.

**Reply by zhaohong-ponyai** [reply]
my test unit is below:
curl  -X PATCH --data '{"rrsets": [{"name": "storage.example.rpz.", "type": "TXT","ttl":60 , "changetype": "REPLACE", "records":[{"content":"\"region=cn\"","disabled":false,"set-ptr":false}] }]}' -
H 'X-API-Key: xxxxyyyyzzzz' http://127.0.0.1:8081/api/v1/servers/localhost/zones/example.rpz.

and got the error :
{"error": "RRset storage.example.rpz. IN TXT: Conflicts with pre-existing RRset"}

and the content of DB is 
![image](https://github.com/PowerDNS/pdns/assets/20126149/f1c1ee6d-9e65-4c3c-94b1-3cb1cdb4f11d)

thank you, looking for your response~

**Reply by Habbie** [reply]
please show `SELECT * FROM records WHERE domain_id=19454`

**Reply by zhaohong-ponyai** [reply]
![image](https://github.com/PowerDNS/pdns/assets/20126149/bcc478d4-bb15-44cc-bd00-58c99582220d)

may be i found the problems
Could it be that my same rrset is a CNAME type, causing it to fail? I tried to add successfully if type is type A, but originally when I used recursor to read the specified db.rpz file, record was also written like this

db.rpz：
```
test.example.rpz IN A 10.0.0.1
storage.gz.example.rpz IN CNAME test.example.rpz
storage.example.rpz IN CNAME storage.gz.example.rpz
storage.example.rpz IN TXT "region=cn"
```

Then use the file recursor.lua, which reads:
`rpzFile("./db.rpz")`

This is the correct resolution of these CNAME and TXT records, and now I save these records to the authoritative, it will not work.

**Reply by Habbie** [reply]
that is not what I asked for. However, it does show the presence of a CNAME, which does not allow any other records to co-exist with it.

**Reply by omoerbeek** [reply]
According to the DNS RFCs, you cannot have a CNAME record together with other types for a name. That is fundamental to DNS. I told that already in the first reply.

**Reply by zhaohong-ponyai** [reply]
Ok, I see, Thank you very much for you patience. And then I'll see how I can fix it.
